packages/core/src/App/Containers/SettingsModal/settings-language.jsx

```diff
@@ -57,10 +58,11 @@ const LanguageSettings = ({ changeCurrentLanguage, current_language, toggleSetti
                         <span
                             id={`dt_settings_${key}_button`}
                             key={key}
-                            onClick={() => {
-                                changeLanguage(key, changeCurrentLanguage);
-                                i18n.changeLanguage(key);
+                            onClick={async () => {
+                                await changeLanguage(key, changeCurrentLanguage);
+                                await i18n.changeLanguage(key);
                                 toggleSettingsModal();
+                                initMoment();
```

Comment:
This will only update the moment's locale when we change the language from UI.

We can also change the language by setting a query param in the URL. - https://app.deriv.com/?lang=es
We should handle that too.

Reply:
@likhith-deriv do you need to provide `lang` variable to `initMoment()` ?

Reply:
This function call should not be present. I will remove it. Thank you